src/frontend/src/utils/condition.rs

```diff
@@ -238,6 +239,16 @@ impl Condition {
                     res.push(ExprImpl::literal_bool(false));
                     break;
                 }
+            } else if !i.has_subquery() {
+                // extract_common_factor requires hash-able ExprImpl
+                let extracted_expr = extract_common_factor(i);
```

Comment:
The name `extracted_expr` sounds like it only contains gcd. 🤔

Reply:
Maybe `extract_common_factor` is not a good name... I'll replace it with a better one if I can.